spec/helpers/codelist_helper_spec.rb

```diff
@@ -122,6 +122,13 @@
       end
     end
 
+    describe "#country_select_options" do
+      it "returns an array of country objects with '' as the first (default) option" do
```

Comment:
This doesn't test that there's an array with anything other than the default option in it.

Reply:
Interesting - Looking at the spec my feeling was that `yaml_to_objects` is tested in that way and that this spec just looks for the default value as this is added later - perhaps the wording is way off?

Reply:
Yeah, perhaps it's the wording. I haven't got a complete picture of what's covered in the test suite.

Reply:
It does look like bulk of the testing is defferred to `yaml_to_objects`. Ideally this and nearby methods that are similar would test the full range of their responses. The fact that another `yaml_to_objects` method is used in the implementation shouldn't change that. I think what's happened is that `yaml_to_objects` was added first and we built new methods around it.

`yaml_to_objects` is still used directly as a public method.

If we wanted to make things clearer here I'd suggest we move the testing for `yaml_to_objects` into a shared example and reused it in each test:

```
describe "#yaml_to_objects" do
  it_behaves_like "returns an array of options as objects"
describe "#country_select_options" do
  it_behaves_like "returns an array of options as objects"
  it "sets a blank default as the first option"
describe "#flow_select_options" do
  it_behaves_like "returns an array of options as objects"
  it "sets ODA as the default option"
```

Given the difficulties in this feature and in getting this pull request merged I think it's reasonable to accept that whist it does touch this code it doesn't introduce a problem. As we have test coverage I'm happy that we leave this refactoring opportunity to the next code that touches it.